applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.cpp

```diff
@@ -359,4 +382,353 @@ void GeometryUtilities::CalculateUnitNormals()
     }
 }
 
+void GeometryUtilities::CalculateGaussianCurvature() {
+
+    KRATOS_TRY
+
+    // check if surface normals are necessary (only for Taubin method)
+    bool normals_necessary = false;
+    for (const auto& rNode : mrModelPart.Nodes()) {
+        std::string technique = this->GetCurvatureTechnique(rNode);
+        if (technique == "Taubin") {
+            normals_necessary = true;
+            break;
+        }
```

Comment:
Will it work??: 
elif (technique == "Meyer") {
             edge_necessary = true;
             break;
         }
if (normals_necessary) this->ComputeUnitSurfaceNormals();
if (edge_necessary) {
         ModelPart& r_edge_model_part = mrModelPart.HasSubModelPart(mrModelPart.Name() + "_edges") ? mrModelPart.GetSubModelPart(mrModelPart.Name() + "_edges") : mrModelPart.CreateSubModelPart(mrModelPart.Name() + "_edges");
         if (r_edge_model_part.Nodes().size() == 0) {
             this->ExtractEdgeNodes(mrModelPart.Name() + "_edges");
         }
     }

Reply:
Unfortunately not, because the loop will break, before it detects a node which has to be calculated by the other technique. 

Let's say we have 1000 nodes with Taubin method and 1000 nodes with Meyer technique. Only the technique of the first node will be detected.